src/box/lua/schema.lua

```diff
@@ -666,6 +666,11 @@ local function normalize_foreign_key(space_id, space_name, fkey, error_prefix,
         local fkey_same_space = (fkey.space == nil or
                                  fkey.space == space_id or
                                  fkey.space == space_name)
+        if fkey_same_space and space_id == nil then
+            box.error(box.error.ILLEGAL_PARAMS, error_prefix ..
+                      "foreign key on same spaced used with spaceless tuple " ..
```

Comment:
Discussed with @locker, decided to simply drop this check, since foreign key constraints don't work anyways.

In order for normalization to work correctly, we pass an `unknown` space name to the normalization function (added a test for this case).